src/Resources/app/administration/src/module/sw-in-app-purchases/component/sw-in-app-purchase-checkout/index.ts

```diff
@@ -118,8 +118,11 @@ export default Shopware.Component.wrapComponentConfig({
                     }
                     break;
                 case 'success':
-                    this.reset();
-                    window.location.reload();
+                    this.inAppPurchasesService.refreshInAppPurchases()
+                        .then(() => {
+                            this.reset();
+                            window.location.reload();
+                        });
```

Comment:
We need to call `this.reset()` before the refresh call, because there is no load state and the buttons are not blocked, so this code would be triggered multiple times when trying to close the modal.
Or we can block the button/modal closing and add another load state, telling the user to wait for the admin to reload - for this we could simply add a little spinner if the state is `success`

Reply:
Third option: we could also throw a notification that an IAP sync is in progress and that the admin will reload any second

Reply:
For MVP I've done the first option. :)